RFC: Process for Creating RFCs

waxlamp opened the pull request:
This is the first official Multinet RFC. It lays out the procedure for creating RFCs, soliciting feedback, and making them official.

TODO:
- [x] require adding the "rendered" link to the PR description

[Rendered](https://github.com/multinet-app/multinet-rfcs/blob/rfc/rfcs/rfcs/0001-rfcs)

waxlamp committed:
Add text for RFC

waxlamp commented:
Looks like GitHub is no longer allowing requests for multiple reviewers without upgrading :angry:

@JackWilb please check this out and review.

Anyone else on the team, please feel free to review as well!

jjnesbitt commented on rfcs/0001-rfcs/proposal.md:
```diff
@@ -0,0 +1,122 @@
+5. Move into your folder: `cd rfcs/0047-my_awesome_proposal`.
+6. Create a `proposal.md` file, and optionally a `README.md` file.
+
+### Writing the RFC in `proposals.md`
```
Is this suppose to be just `proposal.md`?

```suggestion
### Writing the RFC in `proposal.md`
```

jjnesbitt reviewed (commented):
Generally looks good, I've got two main concerns:

1. I worry that if we want to just document something, without it necessarily being an RFC, that we'll be discouraged from placing that here, as we would then need to formalize it. Does it makes sense to have a section for miscellaneous notes, and then a section for RFCs? An example of something that could be represented in a misc note would be the DNS entries. This leads me to my next point...

2. How should we handle sensitive information? I'm not sure if we have much of that, or if there's an interest in keeping that information here, but it would keep our internal docs centralized. If we keep sensitive information here, then we're not able to effectively share this repository with non-collaborators. I'm not sure if there's a desire for that or not, or if we would simply give someone access if they needed it. However it seems to me that these would more represent developer docs, and that anyone trying to develop plugins/spin up their own instance would find this repo useful.

JackWilb commented:
> Looks like GitHub is no longer allowing requests for multiple reviewers without upgrading angry
> 
> @JackWilb please check this out and review.
> 
> Anyone else on the team, please feel free to review as well!

Weird, it let me add myself as a reviewer

JackWilb commented on rfcs/0001-rfcs/proposal.md:
```diff
@@ -0,0 +1,122 @@
+
+1. Clone the RFCs repository with a command such as `git clone https://github.com/multinet-app/multinet-rfcs`.
+2. Create a new branch naming your RFC: `git checkout -b rfc/my-awesome-proposal`.
+3. Push your branch: `git push -u origin rfc/my-awesome-proposal`.
```
Doesn't a branch need new commits before you can push?

waxlamp replied:
Nope!

A branch is just a pointer to a commit. When you create a local branch, it has no corresponding remote branch yet, so pushing simply creates that remote.

I make a habit of doing this so that any local commits I do make are properly shown as being behind the corresponding remote. That's a good reminder to push in order to synchronize myself as soon as my local branch is ahead of the remote (and therefore in potential danger of being lost if my computer explodes).

JackWilb commented on rfcs/0001-rfcs/proposal.md:
```diff
@@ -0,0 +1,122 @@
+- **Metadata.** Insert a section at the very top of the file, delimited by `---`
+  on both sides; include a line reading starting with `author: ` and appending
+  your name, and another starting with `date: ` and the current date in ISO format
+  (i.e., `2020-06-10`).
```
Maybe make this more explicit by making the days marker larger than 12?

waxlamp replied:
You mean to demonstrate that the second number is a month and the third a day?

Good idea.

waxlamp replied:
e82939b3422ef109a98c68876842cb93aea9afbc

JackWilb commented on rfcs/0001-rfcs/proposal.md:
```diff
@@ -0,0 +1,122 @@
+`updated: ` line to the metadata section, including both a date and a name
+separated by a comma. For example:
+
+   updated: 2021-06-10, Roni Choudhury
```
Same suggestion for dates

waxlamp replied:
e82939b3422ef109a98c68876842cb93aea9afbc

waxlamp committed:
Use singular form of `proposal`

Co-authored-by: Jacob Nesbitt <jjnesbitt2@gmail.com>

waxlamp commented:
> Looks like GitHub is no longer allowing requests for multiple reviewers without upgrading 
> 
> @JackWilb please check this out and review.
> 
> Anyone else on the team, please feel free to review as well!

I found out this morning that those limitations are connected to non-paid, private repositories. So I just made this repository public. I don't have any issue with our thinking on Multinet design being public, seeing as it's an open source project to begin with.

waxlamp commented:
> Generally looks good, I've got two main concerns:
> 
> 1. I worry that if we want to just document something, without it necessarily being an RFC, that we'll be discouraged from placing that here, as we would then need to formalize it. Does it makes sense to have a section for miscellaneous notes, and then a section for RFCs? An example of something that could be represented in a misc note would be the DNS entries. This leads me to my next point...

I would argue that documentation that is specifically not an RFC should *not* be placed here.

If you mean that you want a section in this repository for unstructured notes, etc., I might be convinced that's a good idea, but I'd prefer to think we should simply fold that sort of thing into our actual documentation (in the multinet-docs repo).

> 2. How should we handle sensitive information? I'm not sure if we have much of that, or if there's an interest in keeping that information here, but it would keep our internal docs centralized. If we keep sensitive information here, then we're not able to effectively share this repository with non-collaborators. I'm not sure if there's a desire for that or not, or if we would simply give someone access if they needed it. However it seems to me that these would more represent developer docs, and that anyone trying to develop plugins/spin up their own instance would find this repo useful.

We shouldn't keep any sensitive information on GitHub (even in a private repo, commits are not really secret in the sense that we would require for actual sensitive info). I think sensitive info should be kept in whatever safe spots are appropriate: LastPass for passwords, etc.

However, we can still keep our docs centralized by collecting *pointers* to sensitive info in, e.g., our developer docs. The dev docs can mention things like "the Heroku password is kept by Roni Choudhury in a LastPass account; contact him if you need it".

waxlamp committed:
Update example dates to show YYYY-MM-DD format unambiguously

waxlamp committed:
Include requirement for a link to rendered proposal document

jjnesbitt commented:
> If you mean that you want a section in this repository for unstructured notes, etc., I might be convinced that's a good idea, but I'd prefer to think we should simply fold that sort of thing into our actual documentation (in the multinet-docs repo).

Good point, that addresses my concern.

> We shouldn't keep any sensitive information on GitHub (even in a private repo, commits are not really secret in the sense that we would require for actual sensitive info). I think sensitive info should be kept in whatever safe spots are appropriate: LastPass for passwords, etc.
> 
> However, we can still keep our docs centralized by collecting _pointers_ to sensitive info in, e.g., our developer docs. The dev docs can mention things like "the Heroku password is kept by Roni Choudhury in a LastPass account; contact him if you need it".

That sounds good to me :+1:

waxlamp committed:
Use `README.md` in place of `proposal.md`

What used to be `README.md` is now `notes.md`, and the RFC text has been
updated to reflect this change in procedure.

waxlamp committed:
Fix syntax and form of "rendered" link
